### PR TITLE
Calltoaction6

### PR DESCRIPTION
**Pull request by yapmakutu**

### CODE REVIEW

File: editor-components/call_to_action/call_to_action6/call_to_action6.module.scss [hunk 1]
```diff
@@ -1,120 +1,172 @@
...
+                max-width: 80%;
+                color: var(--composer-font-color-primary);
+                padding: 0.938rem;
+                font-size: 1vw;
+                border: 1px solid var(--composer-font-color-secondary);
+                border-radius: 8px;
+                width: 50%;
+                margin-bottom: 8%;
+                margin-top: 8%;
+                //box-shadow: 0 2px 4px rgba(var(--composer-font-color-primary-rgb), 0.1);
```
Comment (Ayazelar) [inline editor-components/call_to_action/call_to_action6/call_to_action6.module.scss:60]:
```suggestion
```

File: editor-components/call_to_action/call_to_action6/call_to_action6.tsx [hunk 2]
```diff
@@ -2,51 +2,73 @@ import * as React from "react";
...
 class CallToAction6Page extends BaseCallToAction {
   constructor(props?: any) {
     super(props, styles);
+
+    this.addProp({
+      type: "image",
+      key: "background_image",
+      displayer: "Background Image",
+      value:
+        "https://c4.wallpaperflare.com/wallpaper/869/24/987/hand-writing-book-man-wallpaper-preview.jpg",
```
Comment (Ayazelar) [inline editor-components/call_to_action/call_to_action6/call_to_action6.tsx:16]: the URL must be spica storage URL

File: editor-components/call_to_action/call_to_action6/call_to_action6.tsx [hunk 3]
```diff
@@ -55,36 +77,73 @@ class CallToAction6Page extends BaseCallToAction {
...
+    const comment = this.getPropValue("comment", { as_string: true });
+
+    const button = this.castToObject<{
+      button_text: JSX.Element;
+      link: string;
+    }>("button");
+
+    const icon = this.getPropValue("icon");
+
+    const isButtonEmpty = icon || this.castToString(button.button_text);
```
Comment (Ayazelar) [inline editor-components/call_to_action/call_to_action6/call_to_action6.tsx:91]: this shouldn't be like     const isButtonEmpty = icon || !this.castToString(button.button_text);

**Review by Ayazelar — CHANGES_REQUESTED** [review]
.

File: editor-components/call_to_action/call_to_action6/call_to_action6.tsx [hunk 4]
```diff
@@ -55,36 +77,73 @@ class CallToAction6Page extends BaseCallToAction {
   }
 
   render() {
+    const title = this.getPropValue("title", { as_string: true });
+    const placeholder = this.getPropValue("placeholder", { as_string: true });
+    const comment = this.getPropValue("comment", { as_string: true });
+
+    const button = this.castToObject<{
+      button_text: JSX.Element;
+      link: string;
```
Comment (dogusmiuw) [inline editor-components/call_to_action/call_to_action6/call_to_action6.tsx:86]: create a type for this.
[please read](https://github.com/Teknodev/composer-tools/blob/main/editor-components/README.md)

**Review by dogusmiuw — CHANGES_REQUESTED** [review]
please handle changes

**Review by Ayazelar — APPROVED** [review]

**Review by dogusmiuw — APPROVED** [review]

**Pull request merged**